### ISSUE TITLE
Log Table: Losing rows when inserting

### ISSUE THREAD

**Issue by giuliaScalaberni** [yandex/ClickHouse#2460]
Hi,
I've got a "Log" engine table with 25612035 rows and everyday I have to insert about 60K rows. To doing this I used the following SQL command
`insert into my_db.complete_table select * from my_db.diff_table;`
But every update I've got a loss near 20 rows, not always the same quote.

If I insert into the complete_table less rows (about 5K new rows) there is no loss.  

If I insert the 60K rows with a CSV inserting like 
`cat file.csv | clickhouse-client --query="INSERT INTO my_db.complete_table FORMAT CSV";` 
there is no loss. 

ClickHouse version is v1.1.54383-stable and it is a single node table distribution installation.

**Comment by giuliaScalaberni** [comment]
Hi, I've tried to reproduce this database with engine MergeTree. The problem still exist. The Database is made of two tables, each with only fields varchar type (a part for one type date for the engine). I use 8192 for index granularity. I insert the rows by an ETL flow by mETL software.
In a PostgreSQL db not on the same VM, I have a view with the same structure as the clickhouse tables. From there I make a SELECT statement on a table every day different, with about 200K rows (diff_table). Then I put this rows in ClickHouse diff_table, doing a Drop and then Create table, so an Insert command. I've tried with a CURL and a JDBC Connection from PG to CH, but the bug still live. When the insert from Postgre to CH diff_table has finished, I do an Insert select * from diff_table into the CH complete_table that every day increase its number of rows. I've noticed that if I put a delay of 15mins between the insert into diff_table and insert into complete_table from diff_table, the loss sometimes doesn't happen. In the log file I see that after 2-3 mins from the first insert, the db starts doing merging operations on diff_table. I suppouse that these merging operations on diff_table affect the second insert on complete_table, getting a loss of data.

**Comment by blinkov** [comment]
@giuliaScalaberni could you please provide step-by-step instruction on how to reproduce this? (preferably on recent ClickHouse release)

**Comment by constantOut** [comment]
Hello guys, I have very similar problem with INSERT FROM SELECT.
I've decided to experiment with partitioning effect on performance so I needed to copy data to partitioned table and found that some data is lost. I've decided to make a test with exactly same schema for the target schema as a source table has. For some dates it's fine, and for some other dates I lose significant fraction of data. What is even more surprising, when I created 3-rd table with the same schema and copied data there - I've lost some more data.

Fraction of table definition:
```ENGINE = ReplicatedCollapsingMergeTree(
  '/clickhouse/tables/{shard}/p_events',
  '{replica}',
  event_date,
  some_composite_primary_key,
  8192,
  sign
)```

Maybe I have to check ClickHouse logs for something particular ?
